staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/library/lists.go

```diff
@@ -101,120 +95,64 @@ var listsLib = &lists{}
 
 type lists struct{}
 
-var paramA = decls.NewTypeParamType("A")
+var paramA = cel.TypeParamType("A")
 
 // CEL typeParams can be used to constraint to a specific trait (e.g. traits.ComparableType) if the 1st operand is the type to constrain.
 // But the functions we need to constrain are <list<paramType>>, not just <paramType>.
-var summableTypes = map[string]*exprpb.Type{"int": decls.Int, "uint": decls.Uint, "double": decls.Double, "duration": decls.Duration}
-var comparableTypes = map[string]*exprpb.Type{"bool": decls.Bool, "int": decls.Int, "uint": decls.Uint, "double": decls.Double,
-	"duration": decls.Duration, "timestamp": decls.Timestamp, "string": decls.String, "bytes": decls.Bytes}
+var summableTypes = map[string]*cel.Type{"int": cel.IntType, "uint": cel.UintType, "double": cel.DoubleType, "duration": cel.DurationType}
```

Comment:
To ensure that the `summableTypes` are declared in a consistent order, you'll want to make this a list of tuples:

```go
[]struct{
   typeName string
   celType *cel.Type
} {
  { typeName: "int", celType: cel.IntType },
  { typeName: "uint", celType: cel.UintType },
  { typeName: "double", celType: cel.DoubleType },
  { typeName: "duration", celType: cel.DurationType },
}
```

Obviously, feel free to use a more appropriate structure for the calls, but the general idea is that you want a deterministic order of declaration.

Reply:
Updated